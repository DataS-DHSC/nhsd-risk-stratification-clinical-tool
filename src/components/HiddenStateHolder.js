import React from 'react';
import PropTypes from 'prop-types';

const HiddenStateHolder = ({ stateFields }) => (
  <>
    {Object.entries(stateFields).map(([key, value]) => (
      <input
        key={`hidden-input-${key}`}
        type="hidden"
        name={key}
        value={value}
      />
    ))}
  </>
);

HiddenStateHolder.propTypes = {
  stateFields: PropTypes.shape().isRequired,
};

export default HiddenStateHolder;
