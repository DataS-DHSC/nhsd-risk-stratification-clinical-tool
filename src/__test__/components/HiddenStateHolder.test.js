import React from 'react';
import { render } from '@testing-library/react';
import HiddenStateHolder from '../../components/HiddenStateHolder';

describe('HiddenStateHolder', () => {
  it('renders input fields as hidden', () => {
    const props = { stateFields: { field1: 'value1', field2: 'value2' } };
    const container = render(<HiddenStateHolder {...props} />);
    expect(container.asFragment()).toMatchSnapshot();
  });
});
