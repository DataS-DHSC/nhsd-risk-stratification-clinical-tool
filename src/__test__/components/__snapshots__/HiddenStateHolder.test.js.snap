// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HiddenStateHolder renders input fields as hidden 1`] = `
<DocumentFragment>
  <input
    name="field1"
    type="hidden"
    value="value1"
  />
  <input
    name="field2"
    type="hidden"
    value="value2"
  />
</DocumentFragment>
`;
